prac_05/wimbledon.py

```diff
@@ -0,0 +1,56 @@
+"""
+CP1404/CP5632 Practical
+Wimbledon
+
+Estimate: 30 minutes
+Actual:   24 minutes
+
+"""
+
+FILENAME = "wimbledon.csv"
+
+
+def main():
+    """ """
+    data = get_data(FILENAME)
+    champions = get_champions(data)
+    countries = list(get_countries(data))
+    countries.sort()
+
+    print("Wimbledon Champions: ")
+    for champions, count in champions.items():
+        print(f"{champions} {count}")
+    print()
+    print(f"These are {len(countries)} countries have won Wimbledon: ")
+    print(",".join(countries))
+
+
+def get_data(filename):
+    data = []
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
+            row = line.strip().split(',')
+            data.append(row)
+    return data
+
+
+def get_champions(data):
+    champions = {}
+    for row in data:
+        champion = row[2]
```

Comment:
better to define magic number as a constant outside the function

Reply:
the magic number should ideally be defined as a constant outside of the function.

Reply:
Define constants outside of functions